src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -833,7 +833,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		// default to SQL if there are no provides or registered resources
 		let provider = this._providers.get(connection.providerName);
 		if (!provider || !provider.properties || !provider.properties.azureResource) {
-			this._logService.warn('Connection providers incorrectly registered. Defaulting to SQL Azure resource,');
+			this._logService.warn(`Connection provider '${connection.providerName}' is incorrectly registered, defaulting to 'SQL' Azure resource. Provider must specify applicable 'azureResource' in 'connectionProvider' configuration.`);
```

Comment:
If every provider is expected to have this then does it make sense to have this just be a warning? Most (if not all) providers are going to be using a different Azure resource so this seems like something that should be an error at the very least - although having some more obvious way to inform us when this happens seems like a good idea to be able to catch this stuff early.

Reply:
@cheenamalhotra have you tested with PGSQL, MySQL and CosmosDB? what type are we expecting for them? the enum type is defined in azdata.d.ts, does it mean we also need to add those types there?

Reply:
MySQL and PostgreSQL providers look good, they have azureResource set to 'OssRdbms', as it specifically applies to these databases.

CosmosDB extension doesn't specify ([code](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/ee541a748cad1d1eed5186e2fa655181a0a845b4/package.json#L131)), which means they continue to default to 'Sql' as the endpoint applies to Azure CosmosDB.

We should, however, let CosmosDB extension team know to specify Azure Resource as 'Sql' to be specific.

Reply:
@cheenamalhotra any reason that CosmosDB should use `Sql`?

Reply:
Although I can't find specific doc, I think Azure CosmosDB supports authenticating to `https://database.windows.net` resource endpoint - but you can let Cosmos DB teams know if it impacts them, as that's what we default to.

Reply:
@languy FYI, could you please confirm that AAD auth has been working fine for CosmosDB?

Reply:
Hi @languy 

I believe you would also be experiencing an issue similar to https://github.com/microsoft/azuredatastudio/issues/21587, when connecting to Cosmos DB as we can see 'azureResource' is missing from [connectionprovider](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/ee541a748cad1d1eed5186e2fa655181a0a845b4/package.json#L131) configuration.
To fix the same, you'd need to make similar changes to add 'Sql' as your 'azureResource'.

We will improve docs and schema as well to improve visibility of this property.

Reply:
As suggested by @Charles-Gagnon in another issue. this week, I'll add `connection provider` to https://github.com/microsoft/azuredatastudio/wiki/Contribution-points and will ask you to add the azureResource property information there.

Reply:
@languy Are you only ever going to be connecting to "Sql" endpoints (e.g. database.windows.net, although that changes depending on what cloud you're talking to)? Or are there other potential endpoints?

Reply:
@cheenamalhotra @Charles-Gagnon @alanrenmsft
How is this `azureResource` supposed to be used? I don't think I am using it (or at least not intentionally).
There is no plan to connect to Sql endpoints: the cosmosdb extension only connects to cosmosdb endpoints or ARM (to retrieve stats or connection string) or mongo databases endpoint.
As long as it can retrieve the `connectionOptions.azureAccount`, `azureTenantId`, `azureResourceId` in order to connect to ARM, it should work.

Reply:
'azureResource' is used to determine the Azure resource endpoint to be used when acquiring access token for Azure MFA auth. If not specified, it defaults to 'Sql' which corresponds to 'https://database.windows.net'. 

It's not used by extension, but is fetched for the registered provider by azurecore extension, based on configuration requested by provider. Resource URLs are defined [here](https://github.com/microsoft/azuredatastudio/blob/b9e8a8747228bae46b6b5d7239e9abcb21c3af72/extensions/azurecore/src/account-provider/providerSettings.ts#L55).

Reply:
Thanks for clarifying. In that case I am definitely using `azureResource` (to fetch the token) and connecting to 'https://database.windows.net'. I also confirm from the logs that it does revert to 'Sql' as `azureResource` isn't currently defined, but I will specify it in the next release per your recommendation.